src/test/kotlin/org/rust/lang/core/type/RsShortTypeInferenceTest.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.type
+
+class RsShortTypeInferenceTest : RsTypificationTestBase() {
+
+    fun `test ignore ref for levels`() = testExpr("""
+        struct S;
+        fn main() {
+            let s = &&&&S;
+            s;
+          //^ &&&&S
+        }
+    """, short = true)
+
+    fun `test basic test`() = testExpr("""
+        struct S<T, U>;
+
+        impl<T, U> S<T, U> {
+            fn wrap<F>(self, f: F) -> S<F, Self> {
+                unimplemented!()
+            }
+        }
+
+        fn main() {
+            let s: S<(), ()> = unimplemented!();
+            let foo = s
+                .wrap(|x: i32| x + 1)
+                .wrap(|x: i32| x + 1)
+                .wrap(|x: i32| x + 1)
+                .wrap(|x: i32| x + 1);
+            foo;
+            //^ S<fn() -> <unknown>, S<fn() -> <unknown>, S<_, _>>>
+        }
+    """, short = true)
```

Comment:
Seems reasonable! Then let's add a test to `org.rust.lang.core.ext.RsPsiStructureTest`, which allows you to assert about the outcome of arbitrary function of PSI element. This way, we won't tie presentation and type inference tests.